pkg/gitops/deploykey/deploykey.go

```diff
@@ -0,0 +1,45 @@
+package deploykey
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/fluxcd/go-git-providers/github"
+	"github.com/fluxcd/go-git-providers/gitprovider"
+	"github.com/pkg/errors"
+	giturls "github.com/whilp/git-urls"
+)
+
+const githubTokenVariable = "GITHUB_TOKEN"
+
+// GetDeployKeyClient provides a DeployKeyClient for the repo at the given URL
+func GetDeployKeyClient(ctx context.Context, url string) (gitprovider.DeployKeyClient, error) {
+	repo, err := giturls.Parse(url)
+	if err != nil {
+		return nil, err
+	}
+	if strings.Contains(repo.Host, "github.com") {
```

Comment:
nit: use `github.DefaultDomain`

Reply:
maybe, for now, return a typed error if this isn't the case, and put this rest of the logic after that in the main flow?

Reply:
ah yeah of course